core/src/executor/evaluate/function.rs

```diff
@@ -410,3 +411,38 @@ pub fn unwrap(name: String, expr: Evaluated<'_>, selector: Evaluated<'_>) -> Res
 pub fn generate_uuid() -> Value {
     Value::Uuid(Uuid::new_v4().as_u128())
 }
+
+pub fn format(name: String, expr: Evaluated<'_>, format: Evaluated<'_>) -> Result<Value> {
+    match expr.try_into()? {
+        Value::Date(expr) => {
+            let format = eval_to_str!(name, format);
+            format_date(expr, format)
+        }
+        Value::Timestamp(expr) => {
+            let format = eval_to_str!(name, format);
+            format_timestamp(expr, format)
+        }
+        _ => Err(EvaluateError::FunctionRequiresFormattableValue(name).into()),
+    }
+}
+
+fn format_date(a: NaiveDate, b: String) -> Result<Value> {
+    let b = b.as_str();
+    let date = chrono::NaiveDate::format(&a, b).to_string();
+    check_valid_specifier(b, date)
+}
+
+fn format_timestamp(a: NaiveDateTime, b: String) -> Result<Value> {
+    let b = b.as_str();
+    let date = chrono::NaiveDateTime::format(&a, b).to_string();
+    check_valid_specifier(b, date)
+}
+
+fn check_valid_specifier(a: &str, b: String) -> Result<Value> {
```

Comment:
The error handling is done by checking the output from `Chrono` and the specifier that was given as an argument of the function. The `Chrono` returns the specifier given when the `NaiveDate` or NaiveDateTIme` is not formattable.